### PR TITLE
Add Configurability - Api Surface section

### PR DESCRIPTION
**Pull request by Evangelink**
Relates to https://github.com/dotnet/roslyn-analyzers/pull/3987

### CODE REVIEW

**Comment by PRMerger19** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by Evangelink** [comment]
Tagging @mavasani for the merge green light

File: docs/code-quality/ca1002.md [hunk 1]
```diff
@@ -49,6 +49,16 @@ To fix a violation of this rule, change the <xref:System.Collections.Generic.Lis
 
 Do not suppress a warning from this rule unless the assembly that raises this warning is not meant to be a reusable library. For example, it would be safe to suppress this warning in a performance-tuned application where a performance benefit was gained from the use of generic lists.
 
+## Configurability
+
+If you're running this rule from [FxCop analyzers](install-fxcop-analyzers.md) (and not with legacy analysis), you can configure which parts of your codebase to run this rule on, based on their accessibility. For example, to specify that the rule should run only against the non-public API surface, add the following key-value pair to an .editorconfig file in your project:
```
Comment (mavasani) [inline docs/code-quality/ca1002.md:54]: Its fine to mention `FxCop analyzers` in this PR for consistency with other rule documentations, but just an FYI that I am going to remove reference to a specific package (FxCop, FxCop analyzers, etc.) from all rule documentations. At that point, CA rules apply to just source analysis, regardless of the shipping vehicle (NuGet package vs inside .NET SDK itself).

File: docs/code-quality/ca1822.md [hunk 2]
```diff
@@ -35,6 +35,16 @@ Mark the member as static (or Shared in [!INCLUDE[vbprvb](../code-quality/includ
 ## When to suppress warnings
 It is safe to suppress a warning from this rule for previously shipped code for which the fix would be a breaking change.
 
+## Configurability
+
+If you're running this rule from [FxCop analyzers](install-fxcop-analyzers.md) (and not with legacy analysis), you can configure which parts of your codebase to run this rule on, based on their accessibility. For example, to specify that the rule should run only against the non-public API surface, add the following key-value pair to an .editorconfig file in your project:
+
+```ini
+dotnet_code_quality.ca1000.api_surface = private, internal
```
Comment (mavasani) [inline docs/code-quality/ca1822.md:43]:
```suggestion
dotnet_code_quality.ca1822.api_surface = private, internal
```

File: docs/code-quality/ca1707.md [hunk 3]
```diff
@@ -43,6 +43,16 @@ Do not suppress warnings for production code. However, it's safe to suppress thi
 
 For well-known methods in Microsoft code that currently use an underscore and cannot be modified, the rule should be suppressed.
 
+## Configurability
+
+If you're running this rule from [FxCop analyzers](install-fxcop-analyzers.md) (and not with legacy analysis), you can configure which parts of your codebase to run this rule on, based on their accessibility. For example, to specify that the rule should run only against the non-public API surface, add the following key-value pair to an .editorconfig file in your project:
+
+```ini
+dotnet_code_quality.ca1000.api_surface = private, internal
```
Comment (mavasani) [inline docs/code-quality/ca1707.md:51]:
```suggestion
dotnet_code_quality.ca1707.api_surface = private, internal
```

File: docs/code-quality/ca1700.md [hunk 4]
```diff
@@ -49,6 +49,16 @@ To fix a violation of this rule, remove or rename the member.
 
 It is safe to suppress a warning from this rule for a member that is currently used or for libraries that have previously shipped.
 
+## Configurability
+
+If you're running this rule from [FxCop analyzers](install-fxcop-analyzers.md) (and not with legacy analysis), you can configure which parts of your codebase to run this rule on, based on their accessibility. For example, to specify that the rule should run only against the non-public API surface, add the following key-value pair to an .editorconfig file in your project:
+
+```ini
+dotnet_code_quality.ca1000.api_surface = private, internal
```
Comment (mavasani) [inline docs/code-quality/ca1700.md:57]:
```suggestion
dotnet_code_quality.ca1700.api_surface = private, internal
```

File: docs/code-quality/ca1046.md [hunk 5]
```diff
@@ -55,6 +55,16 @@ b and a are == ? No
 c and a are == ? Yes
 ```
 
+## Configurability
+
+If you're running this rule from [FxCop analyzers](install-fxcop-analyzers.md) (and not with legacy analysis), you can configure which parts of your codebase to run this rule on, based on their accessibility. For example, to specify that the rule should run only against the non-public API surface, add the following key-value pair to an .editorconfig file in your project:
+
+```ini
+dotnet_code_quality.ca1000.api_surface = private, internal
```
Comment (mavasani) [inline docs/code-quality/ca1046.md:63]:
```suggestion
dotnet_code_quality.ca1046.api_surface = private, internal
```

File: docs/code-quality/ca1002.md [hunk 6]
```diff
@@ -49,6 +49,16 @@ To fix a violation of this rule, change the <xref:System.Collections.Generic.Lis
...
+
+```ini
+dotnet_code_quality.ca1000.api_surface = private, internal
```
Comment (mavasani) [inline docs/code-quality/ca1002.md:57]:
```suggestion
dotnet_code_quality.ca1002.api_surface = private, internal
```

File: docs/code-quality/ca1002.md (hunk 1 above)
Comment (Evangelink) [reply docs/code-quality/ca1002.md:54]: Ok! Let me know if I can help.

**Comment by Mikejo5000** [comment]
#sign-off

**Pull request merged**